### ISSUE TITLE
Infinite training loop

### ISSUE THREAD

**Issue by gandroz** [TensorSpeech/TensorFlowASR#109]
Hi
Have you experienced such an issue ? I got a model training for several days now and it is looping even after when it seems to be finished. 
![image](https://user-images.githubusercontent.com/38106881/104188852-dde11a80-53e7-11eb-8b5a-e9bea089faae.png)
I dont know why the transducer loss keeps updating neither why the evaluation step is performed infinitly ?

**Comment by usimarit** [comment]
@gandroz that's weird, I calculate the total train steps every epoch [here](https://github.com/TensorSpeech/TensorFlowASR/blob/8b9383c5959bd2755535680daa748959ecb9b6ef/tensorflow_asr/runners/base_runners.py#L236) so it must be able to get the total steps [here](https://github.com/TensorSpeech/TensorFlowASR/blob/8b9383c5959bd2755535680daa748959ecb9b6ef/tensorflow_asr/runners/base_runners.py#L88) and check if the training is finished [here](https://github.com/TensorSpeech/TensorFlowASR/blob/8b9383c5959bd2755535680daa748959ecb9b6ef/tensorflow_asr/runners/base_runners.py#L175)

**Comment by gandroz** [comment]
That's weird for sure :) In fact, only the eval steps are performed, there is no training

**Comment by usimarit** [comment]
I see in your picture that it evaluates after each step, did you set the `eval_interval_steps=1`?

**Comment by gandroz** [comment]
no it is set to 500 
![image](https://user-images.githubusercontent.com/38106881/104190671-6e205f00-53ea-11eb-8739-c3c5157528d6.png)

**Comment by gandroz** [comment]
it seems that the total number of steps is corrupted for some reason (see le "?" on the training line)...

**Comment by usimarit** [comment]
@gandroz I found an error 😆 this [while loop](https://github.com/TensorSpeech/TensorFlowASR/blob/8b9383c5959bd2755535680daa748959ecb9b6ef/tensorflow_asr/runners/base_runners.py#L205) breaks only it iterates through the whole training data, then in the last epoch, this [line](https://github.com/TensorSpeech/TensorFlowASR/blob/8b9383c5959bd2755535680daa748959ecb9b6ef/tensorflow_asr/runners/base_runners.py#L334) check and see the step is greater then the total steps so it runs evaluation.

**Comment by gandroz** [comment]
Maybe the issue comes from [here](https://github.com/TensorSpeech/TensorFlowASR/blob/8b9383c5959bd2755535680daa748959ecb9b6ef/tensorflow_asr/runners/base_runners.py#L175). When the `train_steps_per_epoch` is `None` (I dont know why but...), the finished condition is never met.

**Comment by usimarit** [comment]
So comment out the line `(self.total_train_steps and self.steps >= self.total_train_steps)` to fix this 😆

**Comment by gandroz** [comment]
> So comment out the line `(self.total_train_steps and self.steps >= self.total_train_steps)` to fix this

Oh good catch, it loops indefinitly :)
But do you know why the transducer loss keeps changing (on the training line) ?

**Comment by usimarit** [comment]
Anyway, I see the loss still can reduce, I think you can backup the current `lastest.h5` to like `model-41.h5` and continue training model to epoch 45 or 50 😆

**Comment by usimarit** [comment]
It keeps changing because of the mean metrics, may be the next batch results in better loss so it displays smaller, then the next results in worse loss to it displays higher.

**Comment by gandroz** [comment]
hmmm ok... 
![image](https://user-images.githubusercontent.com/38106881/104191991-3a463900-53ec-11eb-8deb-3fdacd566696.png)
I'll stop and relaunch training
